WordPress/gutenberg
# Full Site Editing: Add Support for Templates Default and Custom Titles and Descriptions (JS Side)

##  Description

This PR was originally part of #26636 and it depends on #27036.

- New templates created from the Site Editor sidebar will also automatically use the default template types definitions, and will be saved as `draft`. They will become `published` once saved the first time.
- The default definitions are used across the Site Editor as a fallback for templates without title or excerpt.
- Converted the `getTemplateInfo` utility into the `getTemplateInfo` selector, since it relies on the `core/editor` state.

## How has this been tested?

These test steps take advantage of #27034.
If it's not merged yet, just skip the steps related to the "wp-admin Templates list".

1. Open the Site Editor. Behind the scenes, all theme's and plugins' templates will be converted into new `wp_template` items of status `auto-draft`; there might be duplicates for unrelated reasons.
2. Open the wp-admin Templates list (`/edit.php?post_type=wp_template`).
3. Make sure it shows the following columns: title, slug, description, status.
4. Notice it shows also `auto-draft` templates. Feel free to wipe this list clean by moving all templates to trash, and permanently delete them.
5. Open the Site Editor (if you wiped the templates, now it will generate new ones, hopefully without duplicates).
6. Open the sidebar, and navigate into the Templates menu.
7. Make sure the items use the title and description as provided by the definitions list introduced in this PR.
8. Create a new template with the New Template dropdown (click on `+` in the Templates menu).
9. Make sure the items in the New Template dropdown use the definitions list.
10. Make sure the new template's name is the one provided by the definitions list (it should match the one from the New Template dropdown).
11. Check in the sidebar that the new template shows up (it might be easier to just look in the All Templates menu), it has the correct title and description, and is marked as "[Draft]".
12. Click on "Update Design" (it might be disabled: do some changes to enable it).
13. Review the changes, and make sure the template is listed with the correct title.
14. The template has now been published, so make sure it lost the "[Draft]" label.
15. Open the wp-admin Templates list, and make sure the new template shows up with the right title, description, and status.

## Screenshots

<img width="522" alt="Screenshot 2020-10-26 at 17 44 59" src="https://user-images.githubusercontent.com/2070010/97208552-29a20180-17b3-11eb-892b-753f84693ffe.png">

<img width="276" alt="Screenshot 2020-10-26 at 17 44 41" src="https://user-images.githubusercontent.com/2070010/97208602-3888b400-17b3-11eb-9848-9e6662943acc.png">


## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
